src/Commands/MakeFactoryCommand.php

```diff
@@ -0,0 +1,128 @@
+use KoalaFacade\DiamondConsole\Actions\Stub\CopyStubAction;
+use KoalaFacade\DiamondConsole\Commands\Concerns\HasArguments;
+use KoalaFacade\DiamondConsole\Commands\Concerns\HasOptions;
+use KoalaFacade\DiamondConsole\Commands\Concerns\InteractsWithPath;
+use KoalaFacade\DiamondConsole\DataTransferObjects\CopyStubData;
+use KoalaFacade\DiamondConsole\DataTransferObjects\Filesystem\FilePresentData;
+use KoalaFacade\DiamondConsole\DataTransferObjects\PlaceholderData;
+
+class MakeFactoryCommand extends Command
+{
+    use InteractsWithPath, HasArguments, HasOptions;
+
+    protected $signature = 'diamond:factory {name} {domain} {--force}';
+
+    protected $description = 'create new enum';
+
+    /**
+     * @throws FileNotFoundException
+     */
+    public function handle(): void
+    {
+        $this->info(string: 'Generating factory & interface file to your project');
+
+        $this->resolveGenerateForFactoryInterface();
+
+        $this->resolveGenerateForFactoryConcrete();
+    }
+
+    /**
+     * @throws FileNotFoundException
+     */
+    protected function resolveGenerateForFactoryInterface(): void
+    {
+        $fileName = 'Abstract' . $this->resolveNameArgument() . '.php';
+
+        $namespace = $this->resolveNamespace(
+            identifier: 'Models\\Contracts',
+            domain: 'Shared\\' . $this->resolveDomainArgument(),
+        );
+
+        $destinationPath = $this->resolveNamespaceTarget(namespace: $namespace);
+
+        $filePresent = FilePresentAction::resolve()->execute(
+            data: new FilePresentData(
+                fileName: $fileName,
+                destinationPath: $destinationPath,
+            ),
+            withForce: $this->resolveForceOption()
+        );
+
+        if ($filePresent) {
+            $this->warn(string: $fileName . ' is already exists at ' . $destinationPath);
+
+            return;
+        }
+
+        $placeholders = new PlaceholderData(
+            namespace: $namespace,
+            class: $this->resolveClassNameByFile(name: $fileName)
+        );
+
+        CopyStubAction::resolve()
+            ->execute(
+                data: new CopyStubData(
+                    stubPath: $this->resolvePathForStub(name: 'factory-interface'),
+                    destinationPath: $destinationPath,
+                    fileName: $fileName,
+                    placeholders: $placeholders,
+                )
+            );
+
+        $this->info(string: 'Succeed generate Factory Interface at ' . $destinationPath . '/' . $fileName);
+    }
+
+    /**
+     * @throws FileNotFoundException
+     */
+    protected function resolveGenerateForFactoryConcrete(): void
+    {
+        $fileName = $this->resolveNameArgument() . '.php';
+
+        $namespace = Str::of(string: 'Factories')
+            ->start(prefix: 'Database\\')
+            ->start(prefix: $this->resolveDomainArgument() . '\\')
+            ->start(prefix: $this->resolvePathForInfrastructure() . '\\');
```

Comment:
resolveInfrastructurePath() i think we no longer use `For` naming, the reason is when we use `For` word its meaning we need to pass param, so if you wrote pathForInfra then u need to pass `$path` variable into parameter

Reply:
without `For` naming we can flexible to pass some params or without and more nature naming